src/InodeCache.cpp

```diff
@@ -258,40 +306,21 @@ InodeCache::with_bucket(const Digest& key_digest,
   Util::big_endian_to_int(key_digest.bytes(), hash);
   const uint32_t index = hash % k_num_buckets;
   Bucket* bucket = &m_sr->buckets[index];
-  int err = pthread_mutex_lock(&bucket->mt);
-#ifdef HAVE_PTHREAD_MUTEX_ROBUST
-  if (err == EOWNERDEAD) {
+  const bool broken_lock = spin_lock(bucket->owner_pid, m_self_pid);
+  if (broken_lock) {
     if (m_config.debug()) {
       ++m_sr->errors;
     }
-    err = pthread_mutex_consistent(&bucket->mt);
-    if (err) {
-      LOG(
-        "Can't consolidate stale mutex at index {}: {}", index, strerror(err));
-      LOG_RAW("Consider removing the inode cache file if the problem persists");
-      return false;
-    }
     LOG("Wiping bucket at index {} because of stale mutex", index);
     memset(bucket->entries, 0, sizeof(Bucket::entries));
```

Comment:
Clearing the bucket is no longer safe since the old lock owner, if still alive, could be writing new content to the same bucket simultaneously, thus leaving the bucket in undefined state. So instead we need to wipe the entire cache file by unmapping, unlinking and creating a new one. See InodeCache::drop() and InodeCache::initialize().

Reply:
Done

Reply:
Looks like this conversation is somehow still marked as unresolved, but I can't figure out how to resolve it since it got obsoleted by the force push.

Reply:
Resolved it (I didn't resolve it immediately to check that implementation is right).